src/AzureSqlAction.ts

```diff
@@ -164,13 +164,20 @@ export default class AzureSqlAction {
         let args = '';
 
         switch (inputs.sqlpackageAction) {
-            case SqlPackageAction.Publish: {
-                args += `/Action:Publish /TargetConnectionString:"${inputs.connectionConfig.ConnectionString}" /SourceFile:"${inputs.filePath}"`;
+            case SqlPackageAction.Publish: 
+            case SqlPackageAction.Script:
+            case SqlPackageAction.DeployReport:
+            case SqlPackageAction.DriftReport:
+                args += `/Action:${SqlPackageAction[inputs.sqlpackageAction]} /TargetConnectionString:"${inputs.connectionConfig.ConnectionString}" /SourceFile:"${inputs.filePath}"`;
                 break;
-            }
-            default: {
+            
+                // When action isn't specified, default to Publish
+                case undefined:
+                    args += `/Action:Publish /TargetConnectionString:"${inputs.connectionConfig.ConnectionString}" /SourceFile:"${inputs.filePath}"`;
+                    break;
+
+            default:
                 throw new Error(`Not supported SqlPackage action: '${SqlPackageAction[inputs.sqlpackageAction]}'`);
-            }
         }
 
         if (!!inputs.additionalArguments) {
```

Comment:
what happens if the user doesn't pass in the OutputPath or DeployScriptPath for these new actions?

Reply:
sqlpackage will error out asking for one of those to be specified